gobblin-core/src/main/java/gobblin/converter/filter/AvroFieldsPickConverter.java

```diff
@@ -0,0 +1,182 @@
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.AvroToAvroConverterBase;
+import gobblin.converter.DataConversionException;
+import gobblin.converter.SchemaConversionException;
+import gobblin.converter.SingleRecordIterable;
+import gobblin.util.AvroUtils;
+
+/**
+ * Converts schema and data by choosing only selected fields provided by user.
+ */
+public class AvroFieldsPickConverter extends AvroToAvroConverterBase {
+  private static final Logger LOG = LoggerFactory.getLogger(AvroFieldsPickConverter.class);
+
+  private static final Splitter SPLITTER_ON_COMMA = Splitter.on(',').trimResults().omitEmptyStrings();
+  private static final Joiner JOINER_ON_DOT = Joiner.on('.');
+  private static final Joiner JOINER_ON_COMMA = Joiner.on(',');
+
+  /**
+   * Convert the schema to contain only specified field. This will reuse AvroSchemaFieldRemover by listing fields not specified and remove it
+   * from the schema
+   * 1. Retrieve list of fields from property
+   * 2. Traverse schema and get list of fields to be removed
+   * 3. While traversing also confirm specified fields from property also exist
+   * 4. Convert schema by using AvroSchemaFieldRemover
+   *
+   * Each Avro Record type increments depth and from input depth is represented by '.'. Avro schema is always expected to start with Record type
+   * and first record type is depth 0 and won't be represented by '.'. As it's always expected to start with Record type, it's not necessary to disambiguate.
+   * After first record type, if it reaches another record type, the prefix of the field name will be
+   * "[Record name].".
+   *
+   * Example:
+   * {
+          "namespace": "example.avro",
+          "type": "record",
+          "name": "user",
+          "fields": [
+            {
+              "name": "name",
+              "type": "string"
+            },
+            {
+              "name": "favorite_number",
+              "type": [
+                "int",
+                "null"
+              ]
+            },
+            {
+              "type": "record",
+              "name": "address",
+              "fields": [
+                {
+                  "name": "city",
+                  "type": "string"
+                }
+              ]
+            }
+          ]
+        }
+   * If user wants to only choose name and city, the input parameter should be "name,address.city". Note that it is not user.name as first record is depth zero.
+   * {@inheritDoc}
+   * @see gobblin.converter.AvroToAvroConverterBase#convertSchema(org.apache.avro.Schema, gobblin.configuration.WorkUnitState)
+   */
+  @Override
+  public Schema convertSchema(Schema inputSchema, WorkUnitState workUnit) throws SchemaConversionException {
```

Comment:
I was exploring AvroUtils and was not able to find an easy way to reconstruct Schema mainly because getFieldSchema only provides leaf information which means anyway converter needs to traverse the Schema.
I've used Trie to present dependencies from input and reconstruct the Schema. (Just traversing Schema and add field when it matches was dropped once I learned Schema fields are immutable). Let me know what you think.